common/web/gesture-recognizer/src/pathSegmenter.ts

```diff
@@ -0,0 +1,957 @@
+     *
+     * The f-statistic value must match or exceed its corresponding entry in
+     * the table below, based on the detected DoF (degrees of freedom) for the
+     * 'numerator' and 'denominator' components.  Exceeding the threshold specified
+     * for the p = .100 section indicates less than a 10% probability of the statistic
+     * being tested having arisen by purely random chance.  Failure to exceed even
+     * that implies an unacceptably high chance of being "convenient but meaningless" -
+     * the "null hypothesis", in stats speak.
+     */
+    private static readonly table = [
+      // p = .100
+      [
+        // numerator: 2
+        [
+          // denom: 1-10
+          49.50, 9.00, 5.46, 4.32, 3.78, 3.46, 3.26, 3.11, 3.01, 2.92,
+          // denom: 11-20
+           2.86, 2.81, 2.76, 2.73, 2.70, 2.67, 2.64, 2.62, 2.61, 2.59
+          // we COULD continue with threshold-dropping after that,
+          // but I doubt we'll need it in practicality.
+          // At 30:     2.49
+          // limit -> infinity:  2.30
+        ],
+        // numerator: 3
+        [
+          // denom: 1-10
+          53.59, 9.16, 5.39, 4.19, 3.62, 3.29, 3.07, 2.92, 2.81, 2.73,
+          // denom: 11-20
+           2.66, 2.61, 2.56, 2.52, 2.49, 2.46, 2.44, 2.42, 2.40, 2.38
+          // Likewise, from above.
+          // At 30: 2.28
+          // limit -> infinity: 2.08
+        ]
+      ],
+      // p = .050
+      [
+        // numerator: 2
+        [
+          // denom: 1-10
+          199.5, 19.0, 9.55, 6.94, 5.79, 5.14, 4.74, 4.46, 4.26, 4.10,
+          // denom: 10-20
+           3.98, 3.89, 3.81, 3.74, 3.68, 3.63, 3.59, 3.55, 3.52, 3.49
+          // At 30: 3.32
+          // limit -> infinity: 3.00
+        ],
+        // numerator: 3
+        [
+          // denom: 1-10
+          215.7,19.16, 9.28, 6.59, 5.41, 4.76, 4.35, 4.07, 3.86, 3.71,
+          // denom: 10-20
+           3.59, 3.49, 3.41, 3.34, 3.29, 3.24, 3.20, 3.16, 3.13, 3.10
+          // At 30: 2.92
+          // limit -> infinity: 2.60
+        ]
+      ]
+    ]
+
+    /**
+     * Determines a threshold tier for segmentation based on the segmented
+     * regression f-statistic.
+     * @param statistic
+     * @param numDoF
+     * @param denomDoF
+     *
+     * Returns 0.050 if the statistic indicates a p-value of 0.050 or lower.
+     * Failing that, returns 0.100 if a p-value of 0.100 or lower is indicated.
+     * Otherwise, returns 1.0 - we have no basis to "reject the null hypothesis".
+     */
+    static thresholdTier(statistic: number, numDoF: number, denomDoF: number) {
+      // Former case:  we'd never segment anyway
+      // Latter case:  it's currently wrong to statistically test.
+      //               The F-distribution is not defined for this case.
+      if(numDoF < 2 || denomDoF < 1) {
+        return 1;
+      }
+
+      if(numDoF > 3) {
+        numDoF = 3;
+      }
+
+      const numIndex   = (numDoF > 3 ? 3 : numDoF) - 2;
```

Comment:
You've just ensured that `numDoF` is never `> 3` on line 137. So either remove lines 136-138, or:

```suggestion
      const numIndex   = numDoF - 2;
```